### PR TITLE
Avoid calling methods on `undefined`

### LINKED ISSUE #1133: Aliases number issue in Search bar
**Build:**
Stage

**Affected platforms:**
Windows 10 64 bit
macOS 10.14
Android 11
iOS 14.8

**Browsers tested on:**
Firefox

**Preconditions:**
The user has a few aliases already created.

**Steps to reproduce:**

1. Navigate to https://stage.fxprivaterelay.nonprod.cloudops.mozgcp.net and log in with an existing account.
2. Click in the Search aliases bar and observe the number of aliases.

**Actual result:**
The number of aliases is not displayed. Only the slash is visible.

**Expected result:**
The number of aliases is correctly displayed.

aliases Number in search bar issue

### PR DESCRIPTION
**Pull request by Vinnl**
Fixes #1133.

### CODE REVIEW

File: static/js/filter.js [hunk 1]
```diff
@@ -125,30 +125,36 @@
...
-            .filter(checkbox => checkbox.checked)
-            .map(checkbox => checkbox.dataset.categoryType);
-        setCategoryFilters(categoriesToSet);
-        closeCategoryMenu();
-    });
-    filterCategoryCheckboxes[0].form.addEventListener("reset", () => {
-        setCategoryFilters([]);
-        closeCategoryMenu();
-    });
+    // This element is only available when the user has Premium:
```
Comment (maxxcrawford) [inline static/js/filter.js:143]: This comment put me on a path to test and I discovered a separate issue: https://github.com/mozilla/fx-private-relay/blob/1133-count-in-search-bar/privaterelay/templates/includes/dashboard-filter.html#L25-L26

The `user_profile.has_unlimited` changed to `user_profile.has_premium`. Can you update the logic here? This fix will restore the category filter to be visible to premium users.  

If not fixed in this PR, we can/need to open an issue.

Comment (maxxcrawford) [inline static/js/filter.js:143]: This comment put me on a path to test and I discovered a separate issue: https://github.com/mozilla/fx-private-relay/blob/1133-count-in-search-bar/privaterelay/templates/includes/dashboard-filter.html#L25-L26

The `user_profile.has_unlimited` changed to `user_profile.has_premium`. Can you update the logic here? This fix will restore the category filter to be visible to premium users.  

If not fixed in this PR, we can/need to open an issue.

**Review by maxxcrawford — APPROVED** [review]
Great catch! 

See the nit below. It may warrant changing for this PR if possible.

**Comment by Vinnl** [comment]
The catch is all @Bianca-O's, but great work indeed :)

File: static/js/filter.js (hunk 1 above)
Comment (Vinnl) [reply static/js/filter.js:143]: Nice find! Any idea how/why that happened? @codemist already [fixed](https://github.com/mozilla/fx-private-relay/pull/1130/files#diff-08d75e4f3be8657c70a33e3b353b7c6a6fc3ff628ecbe09c6cfdf952aa0fc880R106) a similar occurrence elsewhere, so it's a bit worrying that this was missed in multiple instances. I'll fix it in a separate PR :)

Comment (Vinnl) [reply static/js/filter.js:143]: PR here: https://github.com/mozilla/fx-private-relay/pull/1152

**Pull request merged**